src/main/webapp/js/omniComment.js

```diff
@@ -217,84 +215,80 @@ $(document).ready(function() {
-                commentListRegionForFeedbackResponse.show();
-                commentListRegionForFeedbackResponse.prev().show();
-                commentListRegionForFeedbackResponse.prev().prev().show();
-                commentListRegionForFeedbackResponse.prev().prev().prev().show();
-                
-                var feedbackQuestion = commentListRegionForFeedbackResponse.closest('.feedback-question-panel');
-                if (feedbackQuestion.find('tr[style*="display: none"]').length !== feedbackQuestion.find('tr').length) {
-                    // if not all responses are hidden within fbQuestion, then show the fbQuestion
-                    feedbackQuestion.show();
-                    
-                    var feedbackSessionPanel = feedbackQuestion.closest('.feedback-session-panel');
-                    var feedbackSessionPanelBody = feedbackQuestion.parent();
-                    if (feedbackSessionPanel.find('div[class="panel panel-info"][style*="display: none"]').length !== feedbackSessionPanel.find('div[class="panel panel-info"]').length) {
-                        // if not all questions are hidden within fbSession, then show the fbsession's body
-                        feedbackSessionPanelBody.show();
-                    }
-                }
-            }
-        }
         // to show student comments (only works for Giver filter)
-        if ($(comment).prop('class').toString().includes(classNameForCommentsInStudentRecords)) {
-            var studentCommentPanel = $(comment).closest('.student-comments-panel');
-            var studentCommentPanelBody = $(comment).parent();
-            // if not all student comments are hidden, then show the student comments panel
-            if (studentCommentPanel.find('div[class*="giver_display-by"][style*="display: none"]').length !== studentCommentPanel.find('div[class*="giver_display-by"]').length) {
-                studentCommentPanelBody.show();
-            }
+        if (commentToShow.hasClass('student-record-comments')) {
+            var studentCommentPanelBody = commentToShow.closest('.student-comments-panel').parent();
+            studentCommentPanelBody.show();
+        } else { // to show feedback question + feedback session panel
+            var commentListRegionForFeedbackResponse = commentToShow.closest('tr');
+            // a fbResponse in instructorCommentsPage consists of the 4 table rows from the
+            // comment list region row to the giver recipient row
+            // The following statement shows the rows between comment list region row and giver recipient row
+            // including the comment list region row and giver recipient row.
+            // See stackoverflow post: http://stackoverflow.com/questions/2770588/jquery-prevuntil-include-start-selector-and-end-selecter
+            commentListRegionForFeedbackResponse.prevUntil('.feedback-response-giver-recipient-row')
```

Comment:
Hmm, yeah it seems this way of doing it isn't that much of an improvement.. will it be possible to generate a class for all the rows that should be hidden and just select the class and hide all of them instead? Would it be too much work? If yes perhaps we can shift it to a new PR :)